src/EntityFrameworkCore.DataEncryption/Internal/ConverterBuilder`1.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.EntityFrameworkCore.DataEncryption.Internal
+{
+    /// <summary>
+    /// A converter builder class which has the model type specified.
+    /// </summary>
+    /// <typeparam name="TModelType">
+    /// The model type.
+    /// </typeparam>
+    public readonly struct ConverterBuilder<TModelType>
+    {
+        internal ConverterBuilder(IEncryptionProvider encryptionProvider, Func<TModelType, byte[]> decoder, Func<Stream, TModelType> encoder)
+        {
+            Debug.Assert(decoder is not null);
+            Debug.Assert(encoder is not null);
+
+            EncryptionProvider = encryptionProvider;
+            Decoder = decoder;
+            Encoder = encoder;
+        }
+
+        private IEncryptionProvider EncryptionProvider { get; }
```

Comment:
If the properties are not exposed, we could just put them as `private readonly` fields in the struct.

Reply:
Good idea. 👍